### PR TITLE
Fix cjs build

### PR DESCRIPTION
**Pull request by joewagner**
The current published cjs build isn't working because the path to the wasm file isn't resolvable.
Here is an example of the failure:
https://github.com/tablelandnetwork/evm-tableland/actions/runs/3998982387/jobs/6862306753

This fixes the problem by detecting if `__dirname` is present to decide what the runtime env is, then building the path differently for esm and cjs.

### CODE REVIEW

File: main.js [hunk 1]
```diff
@@ -8,7 +9,12 @@ import { initSync, __wasm } from "./module.js";
  */
 const init = async (input) => {
   if (typeof input === "undefined") {
-    input = readFileSync("./main.wasm");
+    const wasmPath = typeof __dirname === "undefined"
+      // in esm build path can be relative
+      ? "./main.wasm"
```
Comment (carsonfarmer) [inline main.js:14]: If this is ESM, then we can be more explicit here, and use `new URL("./main.wasm", import.meta.url);`

File: module.js [hunk 2]
```diff
@@ -110,6 +110,9 @@ const initSync = (bytes) => {
  */
 const init = async (input) => {
   if (typeof input === "undefined") {
+    // TODO: using `import.meta` like this does not work in cjs.
```
Comment (carsonfarmer) [inline module.js:113]: Yes, this is correct. We could give it the same treatment we use in `main.js` I think?

**Review by carsonfarmer — CHANGES_REQUESTED** [review]
Yes, thanks for this @joewagner. I was thinkering with this a bit last night, but this approach is much simpler and cleaner than the crazy stuff I was trying. Thanks! I just have one suggestion around being explicit even with the ESM version.

File: main.js (hunk 1 above)
Comment (joewagner) [reply main.js:14]: Brilliant!

File: module.js [hunk 3]
```diff
@@ -110,6 +110,9 @@ const initSync = (bytes) => {
  */
 const init = async (input) => {
   if (typeof input === "undefined") {
+    // NOTE: using `import.meta` like this does not work in cjs.
+    //       However this works in general aside from a warning during build.
+    //       This is because the value of input is always undefined in cjs.
     input = new URL("../main.wasm", import.meta.url);
```
Comment (carsonfarmer) [inline module.js:116]:
```suggestion
    input = new URL("./main.wasm", import.meta.url);
```

**Review by carsonfarmer — APPROVED** [review]
Found a previous mistake (not introduced in this PR)

**Review by carsonfarmer — APPROVED** [review]

**Pull request merged**